src/components/Calculator.js

```diff
@@ -1,44 +1,78 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import calculate from '../logic/calculate';
 // eslint-disable import/no-extraneous-dependencies
 
 // eslint-disable-next-line react/prefer-stateless-function
 class Calculator extends React.Component {
   // eslint-disable-next-line no-useless-constructor
   constructor(props) {
     super(props);
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
+    this.handleClick.bind(this);
   }
 
+  handleClick = (e) => {
+    const buttonName = e.target.textContent;
+    const { total, next, operation } = this.state;
+    const obj = calculate(({ total, next, operation }), buttonName);
+
+    this.setState(() => ({
+      total: obj.total,
+      next: obj.next,
+      operation: obj.operation,
+    }));
+    console.log('buttonName', buttonName);
+    console.log('obj', obj);
```

Comment:
- [ ] please don't push any console.log to the repo they will make your code look unorganised and harder to read.